### DISCUSSION TITLE
Expected argument of type "int", "string" given at property path "xxx".

### DISCUSSION THREAD

**Discussion by azri1521** [symfony/symfony#47294] [Q&A]
Hi,

I got problem mapping entity type integer with form define as TextType.  

entity code

    public function getQuantity(): ?int
    {
        return $this->quantity;
    }

    public function setQuantity(int $quantity): self
    {
        $this->quantity = $quantity;

        return $this;
    }

formType
                ->add('quantity', TextType::class, [
                    'required' => false,
                    'mapped' => true,

if value enter like 'a' it will return Expected argument of type "int", "string" given at property path "quantity".

is there a way to handle value not integer type? constraints only validate when form submit.

Thanks

**Comment by MatTheCat** [comment]
Symfony provides an [`IntegerType`](https://symfony.com/doc/current/reference/forms/types/integer.html) which should fit your use case.

**Reply by azri1521** [reply]
Thanks you. try to use IntegerType but not suitable what I try to do. I manage it with DataTransformerInterface. so far test for server validation works fine.

Thanks again
